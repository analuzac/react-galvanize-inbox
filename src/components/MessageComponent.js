import React from 'react';

function renderLabels(labels) {
  return labels.map(label => {
    return (
      <span className="label label-warning">
        {label}
      </span>
    );
  });
}

// //For the expanded case
// function renderExpanded(expanded) {
//   return expanded.map(bodyMessage => {
//     return (
//       <div className="row message-body">
//         <div className="col-xs-11 col-xs-offset-1">
//           {bodyMessage}
//         </div>
//       </div>
//     );
//   });
// }
//for this: MessageComponent({ expanded, selected, message })

export default function MessageComponent({ selected, message }) {
  return (
    <div>
      <div
        className={
          message.read === true
            ? 'row message read selected'
            : 'row message unread selected'
        }>
        <div className="col-xs-1">
          <div className="row">
            <div className="col-xs-2">
              <input
                type="checkbox"
                checked={selected === true ? 'checked' : null}
              />
            </div>
            <div className="col-xs-2">
              <i
                className={
                  message.starred === true
                    ? 'star fa fa-star'
                    : 'star fa fa-star-o'
                }
              />
            </div>
          </div>
        </div>
        <div className="col-xs-11">
          {renderLabels(message.labels)}
          <a href="www.hi.com">
            {message.subject}
          </a>
        </div>
      </div>
      {/* {expanded === true ? renderExpanded(message.expanded) : null} */}
    </div>
  );
}
