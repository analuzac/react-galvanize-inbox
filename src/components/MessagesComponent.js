import React from 'react';
import MessageComponent from './MessageComponent';

export default function MessagesComponent({ messages, selectedMessageIds }) {
  console.log(messages);
  return (
    <div>
      {messages.map(message =>
        <MessageComponent
          key={message.id}
          selected={selectedMessageIds.forEach(
            selId => (selId === message.id ? true : false)
          )}
          message={message}
        />
      )}
    </div>
  );
}
