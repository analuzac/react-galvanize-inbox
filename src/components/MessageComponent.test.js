import React from 'react';
import { shallow, mount } from 'enzyme';
import MessageComponent from './MessageComponent';
import MessagesComponent from './MessagesComponent';

const onStarMessage = jest.fn();

const onUnstarMessage = jest.fn();

const onSelectMessage = jest.fn();

const onDeselectMessage = jest.fn();

const onMarkAsReadMessage = jest.fn();

const message1 = {
  id: 1,
  subject:
    "You can't input the protocol without calculating the mobile RSS protocol!",
  body:
    'The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog. Junk MTV quiz graced by fox whelps. Bawds jog, flick quartz, vex nymphs. Waltz, bad nymph, for quick jigs vex! Fox nymphs grab quick-jived waltz. Brick quiz whangs jumpy veldt fox. Bright vixens jump; dozy fowl quack. Quick wafting zephyrs vex bold Jim.',
  read: true,
  starred: true,
  labels: ['dev', 'personal']
};

const message2 = {
  id: 1,
  subject:
    "You can't input the protocol without calculating the mobile RSS protocol!",
  body:
    'The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog. Junk MTV quiz graced by fox whelps. Bawds jog, flick quartz, vex nymphs. Waltz, bad nymph, for quick jigs vex! Fox nymphs grab quick-jived waltz. Brick quiz whangs jumpy veldt fox. Bright vixens jump; dozy fowl quack. Quick wafting zephyrs vex bold Jim.',
  read: false,
  starred: false,
  labels: ['dev', 'personal']
};

describe('a shallow test', () => {
  const shallowWrapper_ver1 = shallow(
    <MessageComponent
      selected={message1.id}
      message={message1}
      onStarMessage={onStarMessage}
      onUnstarMessage={onUnstarMessage}
      onSelectMessage={onSelectMessage}
      onDeselectMessage={onDeselectMessage}
      onMarkAsReadMessage={onMarkAsReadMessage}
    />
  );

  const shallowWrapper_ver2 = shallow(
    <MessageComponent
      selected={message2.id}
      message={message2}
      onStarMessage={onStarMessage}
      onUnstarMessage={onUnstarMessage}
      onSelectMessage={onSelectMessage}
      onDeselectMessage={onDeselectMessage}
      onMarkAsReadMessage={onMarkAsReadMessage}
    />
  );

  const fullWrapper_ver1 = mount(
    <MessageComponent
      selected={message1.id}
      message={message1}
      onStarMessage={onStarMessage}
      onUnstarMessage={onUnstarMessage}
      onSelectMessage={onSelectMessage}
      onDeselectMessage={onDeselectMessage}
      onMarkAsReadMessage={onMarkAsReadMessage}
    />
  );

  const fullWrapper_ver2 = mount(
    <MessageComponent
      selected={message2.id}
      message={message2}
      onStarMessage={onStarMessage}
      onUnstarMessage={onUnstarMessage}
      onSelectMessage={onSelectMessage}
      onDeselectMessage={onDeselectMessage}
      onMarkAsReadMessage={onMarkAsReadMessage}
    />
  );

  const fullWrapper_ver3 = mount(
    <MessageComponent
      selected={null}
      message={message2}
      onStarMessage={onStarMessage}
      onUnstarMessage={onUnstarMessage}
      onSelectMessage={onSelectMessage}
      onDeselectMessage={onDeselectMessage}
      onMarkAsReadMessage={onMarkAsReadMessage}
    />
  );

  it('tests that the read CSS class is present when message.read is true', () => {
    expect(shallowWrapper_ver1.find('.read')).toHaveLength(1);
  });

  it('tests that the unread CSS class is present when message.read is false', () => {
    expect(shallowWrapper_ver2.find('.unread')).toHaveLength(1);
  });

  it('tests that the star icon is filled when the message.starred is true', () => {
    expect(shallowWrapper_ver1.find('.star')).toHaveLength(1);
  });

  it('tests that the selected CSS class is present when the selected prop is true', () => {
    expect(shallowWrapper_ver1.find('.something')).toHaveLength(1);
  });

  it('tests that the checkbox is checked when the selected prop is true', () => {
    //console.log(fullWrapper.find('input').debug());
    expect(fullWrapper_ver1.find('input')).toHaveLength(1);
  });

  it('should test if onStarMessage function was fired', () => {
    // const onStarMessage = jest.fn();

    fullWrapper_ver2.find('.star').simulate('click');
    expect(onStarMessage).toHaveBeenCalled();
  });

  it('should test if onUnstarMessage function was fired', () => {
    //const onUnstarMessage = jest.fn();
    console.log(fullWrapper_ver1.find('.star').debug());
    fullWrapper_ver1.find('.star').simulate('click');
    expect(onUnstarMessage).toHaveBeenCalled();
  });

  it('should test if onMarkAsReadMessage function was fired', () => {
    // const onUnstarMessage = jest.fn();

    fullWrapper_ver2.find('.col-xs-11').simulate('click');
    expect(onMarkAsReadMessage).toHaveBeenCalled();
  });

  it('should test if onSelectMessage function was fired', () => {
    // const onUnstarMessage = jest.fn();
    //console.log(fullWrapper_ver2.find('.something').debug());
    fullWrapper_ver2.find('.something').simulate('change');
    expect(onSelectMessage).toHaveBeenCalled();
  });

  it('should test if onDeselectMessage function was fired', () => {
    // const onUnstarMessage = jest.fn();
    //console.log(fullWrapper.find('.something').at(1).debug());
    fullWrapper_ver3.find('.something').simulate('change');
    expect(onDeselectMessage).toHaveBeenCalled();
  });

  it('Subject rendered ', () => {
    //console.log('hi : ' + shallowWrapper2.find('a.msg').text());
    expect(shallowWrapper_ver2.find('a.msg').text()).toEqual(message2.subject);
  });

  expect(shallowWrapper_ver2).toMatchSnapshot();
});
