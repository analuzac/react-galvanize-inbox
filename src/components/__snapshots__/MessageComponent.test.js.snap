// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      className="row message unread selected"
    >
      <div
        className="col-xs-1"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-2"
          >
            <input
              checked={false}
              className="something"
              onChange={[Function]}
              type="checkbox"
            />
          </div>
          <div
            className="col-xs-2"
          >
            <i
              className="star fa fa-star-o"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
      <div
        className="col-xs-11"
        onClick={[Function]}
      >
        <span
          className="label label-warning"
        >
          dev
        </span>
        <span
          className="label label-warning"
        >
          personal
        </span>
        <a
          href="www.hi.com"
        >
          You can't input the protocol without calculating the mobile RSS protocol!
        </a>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        className="row message unread selected"
      >
        <div
          className="col-xs-1"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-2"
            >
              <input
                checked={false}
                className="something"
                onChange={[Function]}
                type="checkbox"
              />
            </div>
            <div
              className="col-xs-2"
            >
              <i
                className="star fa fa-star-o"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="col-xs-11"
          onClick={[Function]}
        >
          <span
            className="label label-warning"
          >
            dev
          </span>
          <span
            className="label label-warning"
          >
            personal
          </span>
          <a
            href="www.hi.com"
          >
            You can't input the protocol without calculating the mobile RSS protocol!
          </a>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MessageComponent
        message={
          Object {
            "body": "The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog. Junk MTV quiz graced by fox whelps. Bawds jog, flick quartz, vex nymphs. Waltz, bad nymph, for quick jigs vex! Fox nymphs grab quick-jived waltz. Brick quiz whangs jumpy veldt fox. Bright vixens jump; dozy fowl quack. Quick wafting zephyrs vex bold Jim.",
            "id": 1,
            "labels": Array [
              "dev",
              "personal",
            ],
            "read": false,
            "starred": false,
            "subject": "You can't input the protocol without calculating the mobile RSS protocol!",
          }
        }
        onDeselectMessage={[Function]}
        onMarkAsReadMessage={[Function]}
        onSelectMessage={[Function]}
        onStarMessage={[Function]}
        onUnstarMessage={[Function]}
        selected={1}
      />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "message": Object {
            "body": "The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog. Junk MTV quiz graced by fox whelps. Bawds jog, flick quartz, vex nymphs. Waltz, bad nymph, for quick jigs vex! Fox nymphs grab quick-jived waltz. Brick quiz whangs jumpy veldt fox. Bright vixens jump; dozy fowl quack. Quick wafting zephyrs vex bold Jim.",
            "id": 1,
            "labels": Array [
              "dev",
              "personal",
            ],
            "read": false,
            "starred": false,
            "subject": "You can't input the protocol without calculating the mobile RSS protocol!",
          },
          "onDeselectMessage": [Function],
          "onMarkAsReadMessage": [Function],
          "onSelectMessage": [Function],
          "onStarMessage": [Function],
          "onUnstarMessage": [Function],
          "selected": 1,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            className="row message unread selected"
          >
            <div
              className="col-xs-1"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-2"
                >
                  <input
                    checked={false}
                    className="something"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
                <div
                  className="col-xs-2"
                >
                  <i
                    className="star fa fa-star-o"
                    onClick={[Function]}
                  />
                </div>
              </div>
            </div>
            <div
              className="col-xs-11"
              onClick={[Function]}
            >
              <span
                className="label label-warning"
              >
                dev
              </span>
              <span
                className="label label-warning"
              >
                personal
              </span>
              <a
                href="www.hi.com"
              >
                You can't input the protocol without calculating the mobile RSS protocol!
              </a>
            </div>
          </div>
        </div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <div
            className="row message unread selected"
          >
            <div
              className="col-xs-1"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-2"
                >
                  <input
                    checked={false}
                    className="something"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
                <div
                  className="col-xs-2"
                >
                  <i
                    className="star fa fa-star-o"
                    onClick={[Function]}
                  />
                </div>
              </div>
            </div>
            <div
              className="col-xs-11"
              onClick={[Function]}
            >
              <span
                className="label label-warning"
              >
                dev
              </span>
              <span
                className="label label-warning"
              >
                personal
              </span>
              <a
                href="www.hi.com"
              >
                You can't input the protocol without calculating the mobile RSS protocol!
              </a>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MessageComponent
    message={
      Object {
        "body": "The quick, brown fox jumps over a lazy dog. DJs flock by when MTV ax quiz prog. Junk MTV quiz graced by fox whelps. Bawds jog, flick quartz, vex nymphs. Waltz, bad nymph, for quick jigs vex! Fox nymphs grab quick-jived waltz. Brick quiz whangs jumpy veldt fox. Bright vixens jump; dozy fowl quack. Quick wafting zephyrs vex bold Jim.",
        "id": 1,
        "labels": Array [
          "dev",
          "personal",
        ],
        "read": false,
        "starred": false,
        "subject": "You can't input the protocol without calculating the mobile RSS protocol!",
      }
    }
    onDeselectMessage={[Function]}
    onMarkAsReadMessage={[Function]}
    onSelectMessage={[Function]}
    onStarMessage={[Function]}
    onUnstarMessage={[Function]}
    selected={1}
  />,
}
`;
